packages/shared-lib/src/core/deviceConfigManifest.ts

```diff
@@ -11,179 +11,59 @@
  * describe some properties to be rendered inside this table
  */
 
+import { TSRActionSchema } from 'timeline-state-resolver-types'
 import { TranslationsBundle } from '../lib/translations'
 
 export interface DeviceConfigManifest {
 	/**
 	 * A description of the config fields
 	 */
-	deviceConfig: ConfigManifestEntry[]
+	deviceConfigSchema: string
 	/**
 	 * If the device has an OAuthFlow (like spreadsheet gw) the instructions for
 	 * getting an authentication token go in here
 	 */
 	deviceOAuthFlow?: DeviceOAuthFlow
 	/**
-	 * A description of the layer mapping config fields
+	 * A description of common properties for each subdevice
 	 */
-	layerMappings?: MappingsManifest
+	subdeviceConfigSchema?: string
 	/**
 	 * A description of how to interact with subdevices
-	 * For now this only includes information on actions but in the future mappings and config options should also live here
 	 */
-	subdeviceManifest?: SubdeviceManifest
+	subdeviceManifest: SubdeviceManifest
 	/**
 	 * Translations for any strings generated by the device that may be shown to the user
 	 */
 	translations?: TranslationsBundle[]
 }
 
-export interface SubdeviceManifest {
-	[deviceType: string | number]: {
+export type SubdeviceManifest<T extends string | number = string | number> = {
+	[deviceType in T]: {
+		/**
+		 * Name to display to users for this device type.
+		 * This will be translated if supplied by the device.
+		 */
+		displayName: string
+		/**
+		 * Stringified JSON schema to use for the settings for this device
+		 */
+		configSchema: string
+		/**
+		 * Playout layer-mappings available for this device type
```

Comment:
I think a little bit more explanation of how this object works would be good? This is a mapping of what strings onto what?

Reply:
This property has changed type to `playoutMappings?: Record<string, JSONBlob<JSONSchema>>`, which hopefully clarifies this?

Reply:
Yeah, I think that's enough clarification.